engine/message/throttle_test.go

```diff
@@ -0,0 +1,98 @@
+package message
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/target/goalert/notification"
+	"testing"
+	"time"
+)
+
+func TestThrottle_Record(t *testing.T) {
+	n := time.Now()
+	trTime := time.Minute
+
+	var cfg = ThrottleConfig{
+		notification.DestTypeSMS: {
+			{Count: 1, Per: trTime},
+		},
+	}
+
+	throttle := NewThrottle(cfg, n, false)
+
+	firstMsg := Message{
+		ID:        "0",
+		Type:      TypeAlertNotification,
+		UserID:    "User A",
+		ServiceID: "Service A",
+		Dest:      notification.Dest{Type: notification.DestTypeSMS, ID: "SMS A"},
+		SentAt:    n,
+	}
+
+	throttle.Record(firstMsg)
+	require.NotNil(t, throttle)
+	require.Equal(t, throttle, &Throttle{
+		cfg:      cfg,
+		ignoreID: false,
+		now:      n,
+		count: map[ThrottleItem]int{
+			{Dest: firstMsg.Dest, BucketDur: trTime}: 1,
+		},
+		cooldown: map[notification.Dest]bool{
+			firstMsg.Dest: true,
+		},
+	})
+
+	secondMsg := Message{
+		ID:        "1",
+		Type:      TypeAlertNotification,
+		UserID:    "User B",
+		ServiceID: "Service B",
+		Dest:      notification.Dest{Type: notification.DestTypeSMS, ID: "SMS B"},
+		SentAt:    n.Add(15 * time.Minute),
+	}
+
+	throttle.Record(secondMsg)
+	require.NotNil(t, throttle)
+	require.Equal(t, throttle, &Throttle{
+		cfg:      cfg,
+		ignoreID: false,
```

Comment:
It's not super helpful to assert on the internal state/implementation of Throttle -- we only care that it works properly.

We can stick with the `InCooldown` test since that uses all relevant public methods (`NewThrottle` `Record` and `InCooldown`) and just rename it to `TestThrottle`.

We could actually change the package to `message_test` at the top of this file to ensure we are only testing the exported methods.